File: ruby/insertion_sort.rb
def insertion_sort(arr)
  arr.each_with_index do |key, i|
    j = i - 1
    j -= 1 while j >= 0 && arr[j] > key
    arr[j + 1] = key
  end
  arr
end

# Example usage
arr = [64, 34, 25, 12, 22, 11, 90]
sorted_arr = insertion_sort(arr)
puts "Sorted array: #{sorted_arr}"

File: javascript/insertion_sort.js
const insertionSort = arr => {
    for (let i = 1; i < arr.length; i++) {
        let key = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
    return arr;
}

// Example usage:
let arr = [64, 34, 25, 12, 22, 11, 90];
console.log("Sorted array: " + insertionSort(arr));
